service/history/historyEngine_test.go

```diff
@@ -5344,42 +5344,40 @@ func copyHistoryEvent(source *historypb.HistoryEvent) *historypb.HistoryEvent {
 	return result
 }
 
-func copyActivityInfo(sourceInfo *persistence.ActivityInfo) *persistence.ActivityInfo {
-	return &persistence.ActivityInfo{
-		Version:                  sourceInfo.Version,
-		ScheduleID:               sourceInfo.ScheduleID,
-		ScheduledEventBatchID:    sourceInfo.ScheduledEventBatchID,
-		ScheduledEvent:           copyHistoryEvent(sourceInfo.ScheduledEvent),
-		StartedID:                sourceInfo.StartedID,
-		StartedEvent:             copyHistoryEvent(sourceInfo.StartedEvent),
-		ActivityID:               sourceInfo.ActivityID,
-		RequestID:                sourceInfo.RequestID,
-		Details:                  proto.Clone(sourceInfo.Details).(*commonpb.Payloads),
-		ScheduledTime:            sourceInfo.ScheduledTime,
-		StartedTime:              sourceInfo.StartedTime,
-		ScheduleToStartTimeout:   sourceInfo.ScheduleToStartTimeout,
-		ScheduleToCloseTimeout:   sourceInfo.ScheduleToCloseTimeout,
-		StartToCloseTimeout:      sourceInfo.StartToCloseTimeout,
-		HeartbeatTimeout:         sourceInfo.HeartbeatTimeout,
-		LastHeartBeatUpdatedTime: sourceInfo.LastHeartBeatUpdatedTime,
-		CancelRequested:          sourceInfo.CancelRequested,
-		CancelRequestID:          sourceInfo.CancelRequestID,
-		TimerTaskStatus:          sourceInfo.TimerTaskStatus,
-		Attempt:                  sourceInfo.Attempt,
-		NamespaceID:              sourceInfo.NamespaceID,
-		StartedIdentity:          sourceInfo.StartedIdentity,
-		TaskQueue:                sourceInfo.TaskQueue,
-		HasRetryPolicy:           sourceInfo.HasRetryPolicy,
-		InitialInterval:          sourceInfo.InitialInterval,
-		BackoffCoefficient:       sourceInfo.BackoffCoefficient,
-		MaximumInterval:          sourceInfo.MaximumInterval,
-		ExpirationTime:           sourceInfo.ExpirationTime,
-		MaximumAttempts:          sourceInfo.MaximumAttempts,
-		NonRetryableErrorTypes:   sourceInfo.NonRetryableErrorTypes,
-		LastFailure:              sourceInfo.LastFailure,
-		LastWorkerIdentity:       sourceInfo.LastWorkerIdentity,
-		// Not written to database - This is used only for deduping heartbeat timer creation
-		LastHeartbeatTimeoutVisibilityInSeconds: sourceInfo.LastHeartbeatTimeoutVisibilityInSeconds,
```

Comment:
Yes, this field was never persisted previously and was used as an in memory cache on the object. It was migrated to a new map in mutableState instead of storing it on ActivityInfos.